File: cpp-sdk/src/DataApiClient.cpp
#include "dataapi/DataApiClient.h"
#include "dataapi/exceptions/DataApiException.h"
#include <stdexcept>

using namespace dataapi::exceptions;

namespace dataapi {

DataApiClient::DataApiClient(const ClientConfig& config, std::shared_ptr<auth::AuthenticationProvider> authProvider)
    : config_(config)
    , authProvider_(authProvider)
    , httpClient_(std::make_shared<http::HttpClient>(config, authProvider))
    , workflowClient_(std::make_unique<client::WorkflowClient>(httpClient_))
    , projectClient_(std::make_unique<client::ProjectClient>(httpClient_))
    , databaseClient_(std::make_unique<client::DatabaseClient>(httpClient_))
    , aiProviderClient_(std::make_unique<client::AiProviderClient>(httpClient_))
    , userClient_(std::make_unique<client::UserClient>(httpClient_)) {
    initialize();
}

DataApiClient::~DataApiClient() = default;

bool DataApiClient::testConnection() {
    try {
        auto response = httpClient_->get("/health");
        return response.statusCode == 200;
    } catch (const std::exception&) {
        return false;
    }
}

ApiVersion DataApiClient::getVersion() {
    auto response = httpClient_->get("/version");
    if (response.statusCode != 200) {
        throw HttpError("Failed to get API version", response.statusCode);
    }
    
    ApiVersion version;
    auto json = Json::parse(response.body);
    version.version = json["version"].get<std::string>();
    version.buildTime = json["buildTime"].get<std::string>();
    version.gitCommit = json["gitCommit"].get<std::string>();
    
    return version;
}

HealthStatus DataApiClient::getHealth() {
    auto response = httpClient_->get("/health");
    if (response.statusCode != 200) {
        throw HttpError("Failed to get health status", response.statusCode);
    }
    
    HealthStatus health;
    auto json = Json::parse(response.body);
    health.status = json["status"].get<std::string>();
    health.message = json["message"].get<std::string>();
    if (json.contains("details")) {
        for (auto& [key, value] : json["details"].items()) {
            health.details[key] = value.get<std::string>();
        }
    }
    
    return health;
}

client::WorkflowClient& DataApiClient::getWorkflowClient() {
    return *workflowClient_;
}

client::ProjectClient& DataApiClient::getProjectClient() {
    return *projectClient_;
}

client::DatabaseClient& DataApiClient::getDatabaseClient() {
    return *databaseClient_;
}

client::AiProviderClient& DataApiClient::getAiProviderClient() {
    return *aiProviderClient_;
}

client::UserClient& DataApiClient::getUserClient() {
    return *userClient_;
}

const ClientConfig& DataApiClient::getConfig() const {
    return config_;
}

std::shared_ptr<auth::AuthenticationProvider> DataApiClient::getAuthProvider() const {
    return authProvider_;
}

std::shared_ptr<http::HttpClient> DataApiClient::getHttpClient() {
    return httpClient_;
}

void DataApiClient::updateConfig(const ClientConfig& newConfig) {
    config_ = newConfig;
    httpClient_ = std::make_shared<http::HttpClient>(config_, authProvider_);
    initialize();
}

void DataApiClient::close() {
    // 清理资源
}

void DataApiClient::initialize() {
    // 初始化逻辑
}

std::unique_ptr<DataApiClient> DataApiClient::createForDevelopment(
    const std::string& apiKey,
    const std::string& baseUrl) {
    ClientConfig config(baseUrl);
    auto authProvider = std::make_shared<auth::ApiKeyAuthProvider>(apiKey);
    return std::make_unique<DataApiClient>(config, authProvider);
}

std::unique_ptr<DataApiClient> DataApiClient::createForProduction(
    const std::string& apiKey,
    const std::string& baseUrl) {
    ClientConfig config(baseUrl);
    auto authProvider = std::make_shared<auth::ApiKeyAuthProvider>(apiKey);
    return std::make_unique<DataApiClient>(config, authProvider);
}

} // namespace dataapi

File: java-sdk/src/main/java/com/owiseman/dataapi/sdk/DataApiClient.java
package com.owiseman.dataapi.sdk;

import com.owiseman.dataapi.sdk.auth.AuthenticationProvider;
import com.owiseman.dataapi.sdk.client.*;
import com.owiseman.dataapi.sdk.config.ClientConfig;
import com.owiseman.dataapi.sdk.exception.DataApiException;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * DataAPI客户端主类
 * 提供对DataAPI后端服务的统一访问接口
 */
public class DataApiClient {
    private static final Logger logger = LoggerFactory.getLogger(DataApiClient.class);
    
    private final ClientConfig config;
    private final AuthenticationProvider authProvider;
    
    // 各种服务客户端
    private final WorkflowClient workflowClient;
    private final ProjectClient projectClient;
    private final DatabaseClient databaseClient;
    private final AiProviderClient aiProviderClient;
    private final UserClient userClient;
    
    /**
     * 构造函数
     * @param config 客户端配置
     * @param authProvider 认证提供者
     */
    public DataApiClient(ClientConfig config, AuthenticationProvider authProvider) {
        this.config = config;
        this.authProvider = authProvider;
        
        // 初始化各种客户端
        this.workflowClient = new WorkflowClient(config, authProvider);
        this.projectClient = new ProjectClient(config, authProvider);
        this.databaseClient = new DatabaseClient(config, authProvider);
        this.aiProviderClient = new AiProviderClient(config, authProvider);
        this.userClient = new UserClient(config, authProvider);
        
        logger.info("DataAPI客户端初始化完成，服务端地址: {}", config.getBaseUrl());
    }
    
    /**
     * 获取工作流客户端
     */
    public WorkflowClient workflows() {
        return workflowClient;
    }
    
    /**
     * 获取项目客户端
     */
    public ProjectClient projects() {
        return projectClient;
    }
    
    /**
     * 获取数据库客户端
     */
    public DatabaseClient databases() {
        return databaseClient;
    }
    
    /**
     * 获取AI提供者客户端
     */
    public AiProviderClient aiProviders() {
        return aiProviderClient;
    }
    
    /**
     * 获取用户客户端
     */
    public UserClient users() {
        return userClient;
    }
    
    /**
     * 测试连接
     */
    public boolean testConnection() {
        try {
            // 尝试获取用户信息来测试连接
            userClient.getCurrentUser();
            return true;
        } catch (DataApiException e) {
            logger.error("连接测试失败", e);
            return false;
        }
    }
    
    /**
     * 关闭客户端，释放资源
     */
    public void close() {
        logger.info("关闭DataAPI客户端");
        // 这里可以添加资源清理逻辑
    }
    
    /**
     * 获取客户端配置
     */
    public ClientConfig getConfig() {
        return config;
    }
}